client/blocks/importer/hooks/use-site-eligible-migration-trial-plan.ts

```diff
@@ -0,0 +1,21 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { isMigrationTrialSite } from 'calypso/sites-dashboard/utils';
+import { useSelector } from 'calypso/state';
+import getSites from 'calypso/state/selectors/get-sites';
+import isA8cTeamMember from 'calypso/state/teams/selectors/is-a8c-team-member';
+
+/**
+ * Allow only one migration trial site per user.
+ * A8c team members are allowed to have more than one.
+ */
+export function useSiteEligibleMigrationTrialPlan(): boolean {
+	const isA8c = useSelector( isA8cTeamMember );
+	const sites = useSelector( getSites );
+	const trialSites = sites.filter( ( site ) => site && isMigrationTrialSite( site ) ) || [];
+
+	if ( ! isEnabled( 'plans/migration-trial' ) ) {
+		return false;
+	}
+
+	return isA8c || ! ( trialSites.length > 0 );
+}
```

Comment:
I think we probably need to have the logic in the backend and output for frontend to use it. There will be few conditions here:
- A8C users are allowed multiple migration trials
- Users with feature config `allow_multiple_migration_trials` enabled are allow multiple migration trials.
- Normal users have ever had a migration trial are not allow to create another one. This also means we will need to include the removed/inactive subscription as well. So `const trialSites = sites.filter( ( site ) => site && isMigrationTrialSite( site ) ) || [];` might not be enough. 

Maybe we can see output another property in `/me` API so we can consume this in frontend. What do you think?

Reply:
Let's park it for now until the backend is ready, thanks @bogiii !